include/aos/iam/certmodule.hpp

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CERTMODULE_HPP_
+#define AOS_CERTMODULE_HPP_
+
+#include "aos/common/crypto.hpp"
+#include "aos/common/time.hpp"
+#include "aos/common/tools/array.hpp"
+#include "aos/common/tools/enum.hpp"
+#include "aos/common/tools/string.hpp"
+#include "aos/iam/config.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/**
+ * Maximum number of module key usages.
+ */
+constexpr auto cModuleKeyUsagesMaxCount = AOS_CONFIG_CERTMODULE_KEY_USAGE_MAX_COUNT;
+
+/**
+ * Certificate type name length.
+ */
+constexpr auto cCertTypeLen = AOS_CONFIG_CERTMODULE_CERT_TYPE_NAME_LEN;
+
+/**
+ * Max number of IAM certificates per module.
+ */
+constexpr auto cCertsPerModule = AOS_CONFIG_CERTHANDLER_MAX_CERTS_PER_MODULE;
+
+/**
+ * Public/private keys generating algorithm
+ */
+class KeyGenAlgorithmType {
+public:
+    enum class Enum { eRSA };
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const sContentTypeStrings[] = {"RSA"};
+        return Array<const char* const>(sContentTypeStrings, ArraySize(sContentTypeStrings));
+    };
+};
+
+using KeyGenAlgorithmEnum = KeyGenAlgorithmType::Enum;
+using KeyGenAlgorithm = EnumStringer<KeyGenAlgorithmType>;
+
+/**
+ * Extended key usage type.
+ */
+class ExtendedKeyUsageType {
+public:
+    enum class Enum { eClientAuth, eServerAuth };
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const sContentTypeStrings[] = {"clientAuth", "serverAuth"};
+        return Array<const char* const>(sContentTypeStrings, ArraySize(sContentTypeStrings));
+    };
+};
+
+using ExtendedKeyUsageEnum = ExtendedKeyUsageType::Enum;
+using ExtendedKeyUsage = EnumStringer<ExtendedKeyUsageType>;
+
+/**
+ * ModuleConfig module configuration.
+ */
+struct ModuleConfig {
+    /**
+     * Key generating algorithm.
+     */
+    KeyGenAlgorithm mKeyGenAlgorithm;
+    /**
+     * Maximum number of certificates for module.
+     */
+    size_t mMaxCertificates;
+    /**
+     * Extra extensions needed for CSR. Current supported values: [clientAuth, serverAuth]
+     */
+    StaticArray<ExtendedKeyUsage, cModuleKeyUsagesMaxCount> mExtendedKeyUsage;
+    /**
+     * Alternative DNS names.
+     */
+    StaticArray<StaticString<crypto::cDnsNameLen>, crypto::cAltDnsNamesCount> mAlternativeNames;
+    /**
+     * Skip certificate chain validation.
+     */
+    bool mSkipValidation;
+};
+
+/**
+ * General certificate information.
+ */
+struct CertInfo {
+    /**
+     * Certificate issuer name.
+     */
+    StaticString<crypto::cCertIssuerLen> mIssuer;
```

Comment:
We should not use string for issuer and serial number. We should use StaticArray<uint8_t> instead.